weblate/trans/tests/test_reports.py

```diff
@@ -384,6 +415,248 @@ def test_counts_inverse_daterange(self) -> None:
             "Error in parameter period: The starting date has to be before the ending date.",
         )
 
+    def test_counts_invalid_sort_order(self) -> None:
+        response = self.get_counts("json", sort_order="editing", follow=True)
+        self.assertContains(
+            response,
+            "Error in parameter sort_order: Select a valid choice. editing is not one of the available choices.",
+        )
+
+    def test_counts_invalid_sort_by(self) -> None:
+        response = self.get_counts("json", sort_by="-", follow=True)
+        self.assertContains(
+            response,
+            "Error in parameter sort_by: Select a valid choice. - is not one of the available choices.",
+        )
+
+    def create_test_sort_data(self) -> None:
+        user1 = User.objects.create(
+            username="customtestuser1",
+            email="custom-weblate-1@example.org",
+            password=TESTPASSWORD,
+            full_name="Weblate Test 1",
+            date_joined=datetime(2025, 1, 1, tzinfo=UTC),
+        )
+        user2 = User.objects.create(
+            username="customtestuser2",
+            email="custom-weblate-2@example.org",
+            password=TESTPASSWORD,
+            full_name="Weblate Test 2",
+            date_joined=datetime(2025, 2, 1, tzinfo=UTC),
+        )
+        user3 = User.objects.create(
+            username="customtestuser3",
+            email="custom-weblate-3@example.org",
+            password=TESTPASSWORD,
+            is_superuser=True,
+            full_name="Weblate Test 3",
+            date_joined=datetime(2025, 3, 1, tzinfo=UTC),
+        )
+
+        self.client.login(username=user1.username, password="testpassword")
```

Comment:
Should use the same constant as when creating:

```suggestion
        self.client.login(username=user1.username, password=TESTPASSWORD)
```

… the same applies to other occurrences below.

Reply:
Hi! I tried this but the login fails in that case because `self.client.login` requires the plaintext password whereas `TESTPASSWORD` is already hashed. I also checked other uses of `self.client.login` in the tests to confirm and they are using the plaintext version as well.